Rubberduck.CodeAnalysis/Inspections/Concrete/DuplicatedAnnotationInspection.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Annotations;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Resources.Inspections;
+
+namespace Rubberduck.Inspections.Concrete
+{
+    public sealed class DuplicatedAnnotationInspection : InspectionBase
+    {
+        public DuplicatedAnnotationInspection(RubberduckParserState state) : base(state)
+        {
+        }
+
+        protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        {
+            var issues = new List<DeclarationInspectionResult>();
+
+            foreach (var declaration in State.AllUserDeclarations)
+            {
+                var duplicateAnnotations = declaration.Annotations
+                    .GroupBy(annotation => annotation.AnnotationType)
+                    .Where(group => !group.First().AllowMultiple && group.Count() > 1 &&
+                                    (group.Key.HasFlag(AnnotationType.ModuleAnnotation) &&
+                                     declaration.DeclarationType.HasFlag(DeclarationType.Module) ||
+                                     group.Key.HasFlag(AnnotationType.MemberAnnotation) &&
+                                     declaration.DeclarationType.HasFlag(DeclarationType.Member)));
```

Comment:
The reason I did this was because a duplicate annotation result would show up for illegal annotations. My reasoning was that if an annotation is illegal, then no inspection results should be shown for such annotations, other than it being illegal.
I understand that this will cause duplicate checks in different inspections, like this one. I was thinking that it would be nice to have a `IsLegal` property on the annotation class and display annotation-specific inspection results only for legal annotations. I'm not sure if that's possible, though.

Reply:
Anyway, if it's OK to show inspection results for illegal annotations, then I'm fine with removing this check and associated tests.